### PR TITLE
fix: transparent effect in California coast theme

### PR DESCRIPTION
**Pull request by AidenLx**
Use `document.body` in favor of `appContainer` since it may not be reliable (`appContainer = this.app.dom.appContainerEl || document.querySelector(".app-container");`

### CODE REVIEW

**Comment by AidenLx** [comment]
Also introduced a new feature to hide left ribbon entirely (except toggle, for course) 
|Before|After|
|:-:|:-:|
|image|image|

File: styles.css [hunk 1]
```diff
@@ -27,7 +26,12 @@
     flex-basis: calc(100% - 4px);
 }
 
-.focus-mode
-    .side-dock-ribbon-action[aria-label="Toggle Focus Mode (Shift + Click to show active pane only)"] {
-    visibility: visible;
+.focus-mode div.workspace-ribbon.mod-left {
+    display: block;
+    position: absolute !important;
```
Comment (ryanpcmcquen) [inline styles.css:34]: Can we achieve this with a more specific selector without using `!important`?

Comment (AidenLx) [reply styles.css:34]: Some themes may change this property themselves

Comment (ryanpcmcquen) [reply styles.css:34]: Yeah, I think we should make it easy to override without needing obscene specificity.

**Comment by ryanpcmcquen** [comment]
This looks strange:
Screen Shot 2021-05-22 at 6 51 38 AM

**Comment by ryanpcmcquen** [comment]
I prefer that the button not move and match the background of everything else, as it currently does:
Screen Shot 2021-05-22 at 6 54 50 AM
Screen Shot 2021-05-22 at 6 54 56 AM
Instead of:
Screen Shot 2021-05-22 at 6 54 17 AM
Screen Shot 2021-05-22 at 6 54 22 AM
If you can resolve that, and remove the `!important`, this will be good to merge.

**Comment by AidenLx** [comment]
The purpose of [ae2d1b](https://github.com/ryanpcmcquen/obsidian-focus-mode/pull/10/commits/ae2d1ba4c33903d66e633161fdddec1056323b34) is for view-content to utilize the space that is occupied by empty ribbon, so in the narrower window, it could be less distracting as follows: 
 
|Before|After|
|:-:|:-:|
|image|image|

Also, If trying to keep the location of the toggle without retaining space of the left ribbon, it could be more distracting to see toggle overlapping with the main text in some cases. For instance, this may happen in preview mode under default theme
image

**Comment by ryanpcmcquen** [comment]
I don't see the need to utilize that space. Having narrower margins is not a helpful win to me.

**Comment by AidenLx** [comment]
Maybe it should be an optional choice for those who prefer the ribbon to be hidden.

**Comment by ryanpcmcquen** [comment]
I have no desire to add options to a plugin that is meant to promote focusing. Configuration is the opposite of focus.

**Comment by AidenLx** [comment]
I can add a option in the setting tab

**Comment by ryanpcmcquen** [comment]
Sorry, I won't approve anything that adds settings to this plugin.

File: styles.css [hunk 2]
```diff
@@ -2,26 +2,36 @@
...
-    .side-dock-ribbon-action[aria-label="Toggle Focus Mode (Shift + Click to show active pane only)"],
-.focus-mode .view-content {
+.focus-mode div.workspace-ribbon.mod-left {
     display: block;
-    visibility: visible;
+    position: absolute !important;
+}
+
+.focus-mode div.workspace-ribbon.mod-left > :not(.side-dock-actions),
+.focus-mode div.workspace-ribbon.mod-left .side-dock-ribbon-action:not([aria-label*="Toggle Focus Mode"]) {
```
Comment (ryanpcmcquen) [inline styles.css:35]: No need to use a wildcard here, we control the label.

**Comment by AidenLx** [comment]
But I wonder why you won't allow different levels of component hiding in the settings? the defaults will remain the same as your preference as before, so if you just ignore this, it won't make any difference.

**Comment by AidenLx** [comment]
Here is a quick demo
https://user-images.githubusercontent.com/31102694/119231430-7e440a00-bb53-11eb-86bd-6965e45e28dd.mp4

**Comment by AidenLx** [comment]
Quick demo: 
https://user-images.githubusercontent.com/31102694/119231773-26a69e00-bb55-11eb-9eb0-31ce4bed1125.mp4

**Comment by AidenLx** [comment]
I think the ability to port the prefs in [obsidian-hider](https://github.com/kepano/obsidian-hider) to Focus Mode is even better.

**Comment by ryanpcmcquen** [comment]
You assume I am capable of ignoring things ...

**Comment by AidenLx** [comment]
Fine, I'll maintain these in my fork for those who might need this feature.

**Review by ryanpcmcquen — APPROVED** [review]

**Pull request merged**